Source/MLX/MLXArray+Bytes.swift

```diff
@@ -0,0 +1,314 @@
+// Copyright © 2024 Apple Inc.
+
+import Cmlx
+import Foundation
+import Metal
+
+// MARK: - Backing / Bytes
+
+extension MLXArray {
+
+    /// Return the dimension where the storage is contiguous.
+    ///
+    /// If this returns 0 then the whole storage is contiguous.  If it returns ndmin + 1 then none of it is contiguous.
+    func contiguousToDimension() -> Int {
+        let shape = self.shape
+        let strides = self.internalStrides
+
+        var expectedStride = 1
+
+        for (dimension, (shape, stride)) in zip(shape, strides).enumerated().reversed() {
+            // as long as the actual strides match the expected (contiguous) strides
+            // the backing is contiguous in these dimensions
+            if stride != expectedStride {
```

Comment:
I like this code a lot! For this purpose I think it is perfect but just as a general comment, the stride can be different than the expected stride and the array can still be contiguous if the shape is 1.

I think you shouldn't change anything here because it will complicate the code without any real benefit. Just to keep in mind and also to reinforce why exposing the strides becomes a tricky business quickly.